### ISSUE TITLE
Bump npm package peer dependencies

### ISSUE THREAD

**Issue by AdamVig** [dependabot/dependabot-core#1242]
Following up on #257, #410, and https://github.com/dependabot/feedback/issues/54.

It would be helpful if Dependabot could automatically bump npm peer dependencies to their latest versions when, for example, a package passes a breaking change boundary. I envision the behavior looking like this:

| dependency version | version in `dependencies` | version in `package-lock.json` | version in `peerDependencies` |
|--------------------|---------------------------|--------------------------------|-------------------------------|
1.0.0 | ^1.0.0 | 1.0.0 | ^1.0.0
1.0.1 | ^1.0.0 | 1.0.1 | ^1.0.0
1.2.0 | ^1.0.0 | 1.2.0 | ^1.0.0
1.2.0 | ^1.0.0 | 1.2.0 | ^1.0.0
2.0.0 | ^2.0.0 | 2.0.0 | ^2.0.0

To summarize: keep the peer dependency version as low as possible, but bump it when the version goes past the range (in the example, `2.0.0` was outside the range `^1.0.0`).

**Comment by GiriB** [comment]
@feelepxyz Do we know the scope of this work? Would it be as easy as updating the field the package json just like we do for dependency or devDependencies? Our teams also want this and we want to see if we can contribute

**Comment by feelepxyz** [comment]
> @feelepxyz Do we know the scope of this work? Would it be as easy as updating the field the package json just like we do for dependency or devDependencies? Our teams also want this and we want to see if we can contribute

@GiriB I think we'll want to switch over from our somewhat hacky use of npm internals to manipulating the dependency tree using arborist which is still under heavy development but used in the upcoming v7 version of npm: https://blog.npmjs.org/post/618653678433435649/npm-v7-series-arborist-deep-dive

This will probably be a significant project as it will change how we do all npm updates and break a bunch of things around error handling where we currently parse the log output from npm directly which we would lose if we switch to arborist.

**Comment by jeffwidman** [comment]
We now use `npm` `8`, which includes `arborist` so if anyone wants to take a crack at this, please go for it. 

I do suggest commenting here with your ideas/questions before investing a bunch of time into a PR, to ensure we're all aligned on how this will work.

**Comment by cobaltt7** [comment]
not stale

**Comment by stefanahman** [comment]
Any updates here?
